internal/fxlog/fxlog.go

```diff
@@ -21,68 +21,121 @@
 package fxlog
 
 import (
-	"fmt"
-	"log"
 	"os"
-	"strings"
 
 	"go.uber.org/fx/internal/fxreflect"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
 var _exit = func() { os.Exit(1) }
 
-// Printer is a formatting printer.
-type Printer interface {
-	Printf(string, ...interface{})
+// Level is the level of logging used by Logger.
+type Level int
+
+const (
+	InfoLevel = iota
+	ErrorLevel
```

Comment:
This should specify the type for the first case.

```suggestion
	InfoLevel Level = iota
	ErrorLevel
```

(ErrorLevel will take type level because Info has type level)